test/framework/cluster.go

```diff
@@ -727,7 +755,7 @@ func (e *ClusterE2ETest) cleanup(f func()) {
 	})
 }
 
-func (e *ClusterE2ETest) cluster() *types.Cluster {
+func (e *ClusterE2ETest) Cluster() *types.Cluster {
```

Comment:
I'm not a huge fan of exporting this. It feels like it probably shouldn't exist anyway, because all it's doing for these tests is wrapping together a cluster name and a kubeconfig file. The former is already exported, and the latter can be determined via kubeconfig.FromClusterName, an already exported function. Exporting it just feels like reinforcing an already iffy behavior.

It seems that you wanted to export it to use it in WaitForResource and related functions, which I think would be a good candidate to move into the e2e framework package, and thereby wouldn't need cluster() to be exported? That or just don't use cluster() would be my preference.